src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -57,8 +57,8 @@ class Definition
     public $innerServiceId;
 
     /**
-     * @param string|null $class     The service class
-     * @param array       $arguments An array of arguments to pass to the service constructor
+     * @param string|Parameter|null $class     The service class
```

Comment:
Should be reverted (same for other similar changes) as this "use case" should be deprecated in 4.4 instead.

Reply:
Yes. Let's put this on hold until the deprecation is merged.

Reply:
#32390